Add 2026 Q2 Best Practices WG quarterly report (#598)
ossf/tac: TI-reports/2026/2026-Q2-BEST-WG.md

# 2026 Q2 Best Practices WG

## Overview

The group continues to be active and is working on several simultaneous projects aligned with our Mission & Vision. There is continued strong demand for best practices and guidance on secure software development and supply chain security.

### Sub-groups

* Guides - [link](https://github.com/ossf/wg-best-practices-os-developers/tree/main/docs)
  * C/C++ Compiler Option Hardening Guide - [link](https://best.openssf.org/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.html)
  * Python Secure Coding Guide - [link](https://best.openssf.org/Secure-Coding-Guide-for-Python/)
* EDU.SIG - [link](https://github.com/ossf/education/)  
* Academic Connections SIG - [link](https://github.com/ossf/education/)
* Memory Safety SIG - [link](https://github.com/ossf/Memory-Safety)  
* OpenSSF Best Practices Badge - [link](https://www.bestpractices.dev/)  
* OpenSSF Scorecard - [link](https://github.com/ossf/scorecard)  
* Best Practices Badge and Developing Secure Software (LFD121) course - [link](https://github.com/ossf/secure-sw-dev-fundamentals)  

> * Web Developer Security Guide (in collaboration with W3C) - [link](https://github.com/w3c-cg/swag/)

Reviewer: Why did you put all the links being a meaningless word "link"?? Just like using "here" as a link anchor this is an anti-pattern. It was much better in the previous reports like: https://github.com/ossf/tac/blob/main/TI-reports/2026/2026-Q1-BEST-WG.md :

```suggestion
* [Guides](https://github.com/ossf/wg-best-practices-os-developers/tree/main/docs)
  * [C/C++ Compiler Option Hardening Guide](https://best.openssf.org/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.html)
  * [Python Secure Coding Guide](https://best.openssf.org/Secure-Coding-Guide-for-Python/)
* [EDU.SIG](https://github.com/ossf/education/)  
* [Academic Connections SIG](https://github.com/ossf/education/)
* [Memory Safety SIG](https://github.com/ossf/Memory-Safety)  
* [OpenSSF Best Practices Badge](https://www.bestpractices.dev/)  
* [OpenSSF Scorecard](https://github.com/ossf/scorecard)  
* [Best Practices Badge and Developing Secure Software (LFD121) course](https://github.com/ossf/secure-sw-dev-fundamentals)  
* [Web Developer Security Guide (in collaboration with W3C)](https://github.com/w3c-cg/swag/)
```

Author: fixed

## Best Practices Badge

### Purpose

* The Open Source Security Foundation (OpenSSF) Best Practices badge is a way for Free/Libre and Open Source Software (FLOSS) projects to show that they follow best practices. Projects can voluntarily self-certify, at no cost, by using this web application to explain how they follow each best practice.

#### Current Status

> * Integrated baseline criteria (levels 1–3) into the badge system. Projects can now complete forms and earn baseline badges, with a transition period ending 2026-06-01 for enforcing new criteria.

Reviewer: Are there 2 different badges? A baseline one and a best practices one?

Reviewer 2: Yes indeed!

Reviewer 2: The two different badges have 2 different URLs.

Reviewer: For the record (in case anyone somehow doesn't know yet ;-) I think that's unfortunate. We should aim to reduce the number of badges to make it easier for people to know what to go for.

* Major performance optimization effort to handle increased load from ML crawlers - switched markdown processor (to Commonmarker/Rust), switched memory allocator to mimalloc, and reduced unnecessary object creation.
* Improved automation so external tools can propose badge fill-ins with user-visible highlights.

> * Best practices badge meets baseline-3 itself: added SPDX SBOM generation, secrets policy, and escalated permissions policy.

Reviewer: That's neat!

### Current Status

* Completed modifications to LFD121 ("Developing Secure Software") to cover 12 areas of the Cybersecurity Skills Framework (CSF) by end of Q1 - tracked via a mapping spreadsheet and submitted as PR #201.
* Top contenders for new courses identified: (1) Embedding AI securely in systems, (2) simplified security architecture course; plan for BEST WG + AI/ML WG collaboration on an AI security guide

### Up Next

* ENISA Risk Framework course in progress (Harald Fischer); Dave Russo stepping back from Edu SIG, looking for new participants.

## Concise Guides

### Purpose

* Artifacts that consolidate BEST practices in OSS software development and management techniques

### Current Status

* The Python Secure Coding Guide has achieved its first release.
* A new concise guide covering upstream collaboration between corporate developers and open source maintainers has kicked off.
* Review and suggest plans for maintenance and alignment to AI-assisted practices for existing guides.

#### Overview of guides

**Guides under active development:**
Python Secure Coding Guide (finalizing)

> Concise guide for upstream collaboration

Reviewer 3: Adding missing bullets

```suggestion
* Python Secure Coding Guide (finalizing)
* Concise guide for upstream collaboration
```

**Completed Guides (under maintenance)**
C/C++ Compiler Annotation Guide
C/C++ Compiler Option Hardening
Security Focused Guide for AI Code Assistant Instructions
Concise Guide for Developing More Secure Software
Concise Guide for Evaluating Open Source Software
npm Best Practices Guide
Source Code Management Platform Configuration Best Practices Guide
Correctly Using Regular Expressions for Secure Input Validation
Simplifying Software Component Updates
The Memory Safety Continuum

> Cyber Resilience Act (CRA) Brief Guide for Open Source Software (OSS) Developers

Reviewer 3: Adding missing bullets

```suggestion
* C/C++ Compiler Annotation Guide
* C/C++ Compiler Option Hardening
* Security Focused Guide for AI Code Assistant Instructions
* Concise Guide for Developing More Secure Software
* Concise Guide for Evaluating Open Source Software
* npm Best Practices Guide
* Source Code Management Platform Configuration Best Practices Guide
* Correctly Using Regular Expressions for Secure Input Validation
* Simplifying Software Component Updates
* The Memory Safety Continuum
* Cyber Resilience Act (CRA) Brief Guide for Open Source Software (OSS) Developers
```

Besides a description of each coding pattern, the guide includes executable code examples for each rule, which allow for an in-depth understanding of each pattern.

### Current Status

* First release achieved and approved. Currently applying the last finishing touches.

### Up Next

* Announcing the first release via a blog post.

## Memory Safety SIG

### Purpose

* The Memory Safety SIG is a group working within the OpenSSF's Best Practices Working Group formed to understand and reduce memory safety vulnerabilities in OSS.

### Current Status

* The group decided that it was time to fold the activity back to the Best Practices Working Group and to continue the existing collaboration through the WG's channels and will not have a separate meetings or a separate mailing list.

### Up Next

* Continue the ongoing scorecard and guides work through the Best Practices Working Group.

## OpenSSF Scorecard

### Purpose

* Automate analysis and trust decisions on the security posture of open source projects.  
* Use this data to proactively improve the security posture of the critical projects the world depends on.

### Current Status

> * 2026 roadmap defined (PR #4952)

Reviewer: Can you please add a link to the PR?

Author: fixed

> * 2026 roadmap defined (PR #4952)

Reviewer 4: ```suggestion
* [2026 roadmap](https://github.com/ossf/scorecard/pull/4952) defined
```

Minor: the automatic link doesn't work since it's in a different repo.

### Up Next

* Roadmap implementation.

## Web Developer Security Guide

### Purpose

* Develop security best practice and guidelines specifically aimed at web developers.

### Current Status

* Published the SWAG security guidelines as a W3C Community Group draft report. Working on getting it at a stable URL.
Closed the web developer survey and drafted a blog post with findings. Key insight: web devs are generalists with surprisingly high security feature adoption. Also contributed a fetch Metadata guide to MDN and reviewed session management docs.

> Reviewing the 400+ page ETSI draft on browser standards related to CRA; working on a threat modelling guide for web developers (MDN PR #42980).

Reviewer: Same here: please add a link to the PR.

Author: fixed

### Up Next

* Continue working on a threat modelling [guide for web developers](https://github.com/mdn/content/pull/42980).

> ## Questions/Issues for the TAC

Reviewer 3: To @lehors' comment about the missing information regarding funding: 

```suggestion
## Funding requests and updates

The WG briefly discussed applying for funding of a technical writer to support the group with reviewing, finalizing, and maintaining the various guides we host. However, there currently no concrete funding request in the pipeline.

The WG currently does not receive funding yet.

## Questions/Issues for the TAC
```
